### ISSUE TITLE
method save of class BasecampConfig fails if field filepath is None

### ISSUE THREAD

**Issue by timecamp-analityk** [phistrom/basecampy3#14]
method save of class BasecampConfig fails if field filepath is None
this happens if object Basecamp3 is created from constructor without a command line configuration
recommend modification so method ignores saving if filepath is None.

```    def save(self, filepath=None):
        if filepath is None:
            filepath = self.filepath
        if filepath is not None:
            try:
[...]```

**Comment by timecamp-analityk** [comment]
proposed fix in  #15

**Comment by phistrom** [comment]
Thanks for the issue and pull request! I've actually gone ahead and refactored `BasecampConfig` to be an abstract class and added two concrete subclasses of it.

Now as of version 0.3.0, when you pass arguments directly into the `Basecamp3` constructor, a `BasecampMemoryConfig` object is created. Calling `save()` or `read()` on this object silently does nothing.

If you pass no arguments, the `Basecamp3` object will look for a configuration file in a few locations. If it finds one, it creates a `BasecampFileConfig` object to represent it. As access tokens expire and the refresh token is used to obtain new ones, the `save()` method you reference above will commit it back to the file as normal.
